include/pybind11/eigen.h

```diff
@@ -453,15 +543,55 @@ struct type_caster<
             // We need to copy: If we need a mutable reference, or we're not supposed to convert
```

Comment:
BTW Can this `if` statement be joined to the one above via an `else`?

Reply:
I'm not sure if this can be done, `need_copy` is being changed inside `if(!need_copy)`

Reply:
Ah, missed that!